contrib/submit-job-v2/src/App/ProtocolForm.tsx

```diff
@@ -532,13 +571,70 @@ export default class ProtocolForm extends React.Component<IProtocolProps, IProto
     });
   }
 
+  private addTensorBoardConfig = () => {
+    if (this.state.needTensorBoard) {
+      if (this.state.logDir !== undefined) {
+        const protocol = yaml.safeLoad(this.state.protocolYAML);
+        const randomStr = Math.random().toString(36).slice(-8);
+        const tensorBoardName = "TensorBoard_" + randomStr;
+        const tensorBoardImage = "tensorBoardImage_" + randomStr;
+        const tensorBoardPort = "tensorBoardPort_" + randomStr;
+        const prerequisites = "prerequisites";
+        let length = 0;
+        if (protocol.hasOwnProperty("prerequisites")) {
+          length = protocol.prerequisites.length;
+        } else {
+          protocol[prerequisites] = [];
+        }
+        protocol.prerequisites[length] = {
+          protocolVersion: 2,
+          name: tensorBoardImage,
+          type: "dockerimage",
+          version: "1.0 - r1.4",
+          contributor: "OpenPAI",
+          uri: "openpai/pai.example.tensorflow",
+        };
+        const portList = " --port=$PAI_CONTAINER_HOST_tensorboard_PORT_LIST";
+        protocol.taskRoles[tensorBoardName] = {
+          instances: 1,
+          completion: {
+            minFailedInstances: 1,
+            minSucceededInstances: null,
+          },
+          taskRetryCount: 0,
+          dockerImage: tensorBoardImage,
+          resourcePerInstance:
+          {
+            cpu: 1,
+            memoryMB: 512,
+            gpu: 0,
+            ports: {},
+          },
+          commands: ["tensorboard --logdir = " + this.state.logDir + portList],
+        };
+        protocol.taskRoles[tensorBoardName].resourcePerInstance.ports[tensorBoardPort] = 1;
+        protocol.extras = { tensorBoardStr: randomStr };
+        this.state.protocolYAML = yaml.safeDump(protocol);
+      } else {
+        alert("Please input the tensorboard log directory!");
+      }
+    }
+  }
+
   private submitProtocol = async (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
     event.preventDefault();
     if (!this.state.protocolYAML) {
       return;
     }
+    const protocolYAMLBackup = this.state.protocolYAML;
+    this.addTensorBoardConfig();
```

Comment:
what if `logDir` is empty after alert?

Reply:
I will check logDir in this function rather than addTensorBoardConfig, Thanks.